src/System.Management.Automation/engine/CommandCompletion/CompletionCompleters.cs

```diff
@@ -1770,28 +1770,35 @@ internal static string ConcatenateStringPathArguments(CommandElementAst stringAs
             {
                 RemoveLastNullCompletionResult(result);
 
-                string enumString = LanguagePrimitives.EnumSingleTypeConverter.EnumValues(parameterType);
-                string separator = CultureInfo.CurrentUICulture.TextInfo.ListSeparator;
-                string[] enumArray = enumString.Split(separator, StringSplitOptions.RemoveEmptyEntries);
+                IList<Enum> enumValues = LanguagePrimitives.EnumSingleTypeConverter.GetEnumValues(parameterType).Cast<Enum>().ToList();
```

Comment:
I still think we have no need to cast because ValidateElement parameter type is _object_.
Also we have no need to collect filtered list.
```suggestion
                IEnumerable enumValues = LanguagePrimitives.EnumSingleTypeConverter.GetEnumValues(parameterType);
```

Reply:
True. Just preferred generic in general to get a typed result in future use of this helper, but I don't currently depend on type so I'll drop it.